### ISSUE TITLE
Compiler Warnings in combination with `doclicence` package

### ISSUE THREAD

**Issue by BaumiCoder** [se2p/se2thesis#17]
Using the [doclicense](https://ctan.org/pkg/doclicense) package with `se2thesis` create some warnings.
With `scrbook`, the base of `se2thesis`, there are no warnings.

I don't know if the warnings trigger any problems. I does not encounter any problems with the `doclicense`, which only occurs with `se2thesis`.
## Example
```tex
\documentclass{se2thesis}

\usepackage{hyperref}
\usepackage[type={CC},modifier={by},version={4.0},lang={english}]{doclicense}


\begin{document}
  Hello
\end{document}
```
### Warnings
```
Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\group_begin:' on input line 7.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\hyphenpenalty' on input line 7.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\group_end:' on input line 7.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\group_begin:' on input line 7.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\hyphenpenalty' on input line 7.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `\group_end:' on input line 7.
```
### Enviroment
- Latexmk, John Collins, 7 Apr. 2024. Version 4.85
- LuaHBTeX, Version 1.18.0 (TeX Live 2024)
- Document Class: se2thesis 2024-05-27 v4.0.3 A thesis class for the Chair of Software Engineering II
- Document Class: scrbook 2023/07/07 v3.41 KOMA-Script document class (book)
- Package: doclicense 2023/10/01 v3.3.0 Support for putting documents under a license

**Comment by BaumiCoder** [comment]
A similar [issue at doclicense](https://github.com/ypid/latex-packages/issues/80) was related with a wrong package loading order between `csquote` and `hyperref`. The first one is loaded by `se2thesis`. I don't know when such automatic loads from the class are performed or how to control the position for this in the loading order.

**Comment by stephanlukasczyk** [comment]
Package load order is sometimes painful, to be honest.  First, `se2thesis` does *not* load `csquotes`; however, `se2packages` loads `csquotes` by default (and you're probably using that package).  This can be prevented by `\usepackage[csquotes=false]{se2packages}`.

I am, however, not sure whether this really solves your issue.  I've looked into the code of [`doclicense`](https://github.com/ypid/latex-packages/blob/main/doclicense/doclicense.dtx), which loads `csquotes` automatically at the end of the preamble, if it has not been loaded ([see](https://github.com/ypid/latex-packages/blob/c170a297ea2a227c044df1c85d1b00100323ea9f/doclicense/doclicense.dtx#L709-L735)).

To be honest, I do not have a good suggestion here; especially `hyperref` is a package that causes interaction troubles, because it does heavy lifting and patching of large parts of the LaTeX internals.  That's the reason why it is usually recommended to load `hyperref` last with very few exceptions (`cleveref` is one of the exceptions).

The most important question to me is: even though you get warnings in the log, does everything work as expected, i.e., is the generated PDF looking the way you expected it?

**Comment by BaumiCoder** [comment]
Thank you for this explanation. The example does not use `se2packages` explicitly, but maybe implicitly through `se2thesis`.  Explicitly using the package with `csquotes=false` does not remove the warnings. `se2packages` or `se2thesis` still seems to influence something here.

I cannot encounter any unexpected behavior in the generated PDF. The warning seems to related to the metadata, because it does not have a suitable line in the source code (`\begin{document}`). The commands from `doclicense` seems to work correctly and the metadata, which is created by `doclicense`, also not change (compared to an example with `scrbook` instead of `se2thesis`).

**Comment by stephanlukasczyk** [comment]
> The example does not use `se2packages` explicitly, but maybe implicitly through `se2thesis`. Explicitly using the package with csquotes=false does not remove the warnings. se2packages or se2thesis still seems to influence something here.

No, it is not loaded implicitly.  Note, however, that `doclicense` loads `csquotes` implicitly at the end of the preamble if it has not been loaded before (see the link in my previous post).

> I cannot encounter any unexpected behavior in the generated PDF. The warning seems to related to the metadata, because it does not have a suitable line in the source code (`\begin{document}`). The commands from `doclicense` seems to work correctly and the metadata, which is created by `doclicense`, also not change (compared to an example with `scrbook` instead of `se2thesis`).

Thanks for reporting this back.  The warnings are ugly, yes, but since the result is looking fine, I'm closing this issue.

**Comment by BaumiCoder** [comment]
I now came a cross an issue in the metadata of documents, which are using se2thesis, which is triggered by the use of `doclicence`. The `author` (German `Autor`) attribute of the PDF gets the prefix `=100000`. Here is the metadata information in PDF.js (inside VSCodium):

<img width="360" height="371" alt="Image" src="https://github.com/user-attachments/assets/d5da5425-8ed1-43e5-820c-cd4c7110934e" />

The PDF viewer Okular does not even show the author field if it has this strange prefix.

The following code block contains the diff for the example document to reproduce the issue.

```diff
diff --git a/examples/se2thesis-master-thesis-example.tex b/examples/se2thesis-master-thesis-example.tex
index fececce..3d918b9 100644
--- a/examples/se2thesis-master-thesis-example.tex
+++ b/examples/se2thesis-master-thesis-example.tex
@@ -3,7 +3,6 @@
   open=right,%
   twoside=true,%
   paper=a4,%
-  logofile={logo.pdf},%
   thesistype=master,%
   UKenglish,%
 ]{se2thesis}
@@ -31,7 +30,8 @@
 \end{filecontents}
 \addbibresource{\jobname.bib}
 
-\usepackage{hyperref}
+\usepackage[pdfusetitle]{hyperref}
+\usepackage[type=CC,modifier=by-sa,version=4.0]{doclicense}
 
 \author{Stephan Lukasczyk}
 \title{A Master Thesis Example Document}
```

The `pdfusetitle` property for `hyperref` adds the title and author to the PDF metadata. Only if you also add the `doclicense` package, the prefix is added to the author attribute.
(It seems that the newer TeX version did not introduce that issue, as my bachelor's thesis PDF (build in 2024) also contains this issue, but I had not recognized it until now.)

Therefore, the issue should be reopend, but I have not the permission to do so.

**Comment by stephanlukasczyk** [comment]
To resolve the issue, you could set the PDF properties using
```latex
\usepackage{hyperref}
\hypersetup{
  pdfauthor = {Author Name},
  pdftitle = {Document Title},
}
```

**Comment by b-fein** [comment]
The `=100000` is most likely this one from the redefinition of `\@author`

https://github.com/se2p/se2thesis/blob/a0c97d2a5547c37f7233141495ba6b79d79bd99a/se2thesis.dtx#L1174

**Comment by BaumiCoder** [comment]
> To resolve the issue, you could set the PDF properties using
> 
> \usepackage{hyperref}
> \hypersetup{
>   pdfauthor = {Author Name},
>   pdftitle = {Document Title},
> }

I tried this instead of `\usepackage[pdfusetitle]{hyperref}` in the example document. Now, the author attribute stays empty (also in PDF.js) if the `doclicence` is in use.

**Comment by stephanlukasczyk** [comment]
I have fixed the author issue, thanks to @b-fein 's [comment](https://github.com/se2p/se2thesis/issues/17#issuecomment-4082511888), and released [v4.4.1](https://github.com/se2p/se2thesis/releases/tag/v4.4.1).  I have also uploaded the released version to CTAN, give it a couple of days to get installed and populated throughout the CTAN network.

**Comment by BaumiCoder** [comment]
I can confirm that it is fixed with `se2thesis 2026-04-03 v4.4.1`, tested with container image `registry.gitlab.com/islandoftex/images/texlive:TL2026-2026-04-19-full`.
